### ISSUE TITLE
Gateway Class status should show supported features

### ISSUE THREAD

**Issue by LiorLieberman** [kubernetes-sigs/gateway-api#2162]
**What would you like to be added**:
Improve UX by enabling them to easily see what features their installed GatewayClass support. 

**Why this is needed**:
With the evolution of the Conformance tests suite and features, we want to;

1. Enable users easily see what features the implementation support
1. Improving UX and potentially provide clear errors when a user is trying to use a feature the installed GWC doesn't support

cc: @shaneutt

**Comment by shaneutt** [comment]
Sounds good.

/triage accepted

While it would be really nice to have for GA I don't think it needs to be _required_ for milestone `v1.0.0`. We're open to help from the community on this one, but we would ask that `v1.0.0` issues be prioritized _first_ unless this issue is the _only_ thing you really wanna work on.

/help

**Comment by LiorLieberman** [comment]
This is the enhancement issue for https://github.com/kubernetes-sigs/gateway-api/pull/2163

you already added it to `v1.0.0` milestone. 

/remove help
/assign @LiorLieberman

**Comment by robscott** [comment]
Leaving open until this reaches GA.

/reopen

**Comment by LiorLieberman** [comment]
PRs in flight: 
https://github.com/envoyproxy/gateway/pull/2491
https://github.com/kubernetes-sigs/ingress2gateway/issues/125

**Comment by shaneutt** [comment]
We discussed this on a Gateway API sync: it may be helpful if we package in the conformance part in the initial delivery of this, as that might be a helpful way to encourage implementations.

**Comment by mikemorris** [comment]
@robscott Have you thought about how GKE might handle this (given the whole lack of in-cluster GatewayClass resources approach we share on AKS) if we try to go in the direction of using this advertisement of supported functionality to configure the conformance suite?

/cc @snehachhabria

**Comment by snehachhabria** [comment]
> ~@robscott Have you thought about how GKE might handle this (given the whole lack of in-cluster GatewayClass resources approach we share on AKS) if we try to go in the direction of using this advertisement of supported functionality to configure the conformance suite?~
> 
> **EDIT:** I was wrong, we do actually deploy a GatewayClass into the cluster (just don't supported a user-defined GatewayClass), so we shouldn't be blocked here.
> 
> /cc @snehachhabria

that's right, we do have predefined in-cluster gatewayclasses, so no issues form our end.

**Comment by arkodg** [comment]
thanks @levikobi for adding support for this in Envoy Gateway ! sharing some thoughts after seeing it in action
* biggest value prop here is for users who can figure out which features the implementation supports, however just the name is not enough here, and a URL to the feature page would be great here, to make this more actionable and avoid searching how to use the feature
* the current flat structure if going to get hard to consume for the user soon once we add more support for more things and tests
* implementations may choose to use this field to highlight any extra features they support, to increase feature usage

**Comment by rikatz** [comment]
/close
Lior told me this can be closed
